docs/apache-airflow/concepts/dags.rst

```diff
@@ -413,7 +411,13 @@ You can also combine this with the :ref:`concepts:depends-on-past` functionality
         )
 
         run_this_first = EmptyOperator(task_id="run_this_first", dag=dag)
-        branching = BranchPythonOperator(task_id="branching", dag=dag, python_callable=lambda: "branch_a")
+
+        @task.branch(task_id="branching")
+        def branching():
+            return "branch_a"
+
+        
+        branching = branching()
```

Comment:
```suggestion
        @task.branch(task_id="branching")
        def do_branching():
            return "branch_a"

        
        branching = do_branching()
```

Using the same name for the function and task should be discouraged

Reply:
@uranusjr Is there any specific reason why function and task should not have the same name?

Reply:
One reason is that Mypy would complain since the types of the two are not compatible.